aws/resource_aws_globalaccelerator_listener.go

```diff
@@ -244,17 +236,10 @@ func resourceAwsGlobalAcceleratorListenerDelete(d *schema.ResourceData, meta int
 	}
 
 	// Deleting a listener triggers the accelerator to change status to InPending
-	stateConf := &resource.StateChangeConf{
-		Pending: []string{globalaccelerator.AcceleratorStatusInProgress},
-		Target:  []string{globalaccelerator.AcceleratorStatusDeployed},
-		Refresh: resourceAwsGlobalAcceleratorAcceleratorStateRefreshFunc(conn, d.Get("accelerator_arn").(string)),
-		Timeout: 5 * time.Minute,
-	}
-
-	log.Printf("[DEBUG] Waiting for Global Accelerator listener (%s) deletion", d.Id())
-	_, err = stateConf.WaitForState()
+	// }
+	err = resourceAwsGlobalAcceleratorAcceleratorWaitForDeployedState(conn, d.Get("accelerator_arn").(string))
 	if err != nil {
-		return fmt.Errorf("Error waiting for Global Accelerator listener (%s) deletion: %s", d.Id(), err)
+		return err
```

Comment:
Same here regarding continuing to return the error context:

```suggestion
		return fmt.Errorf("error waiting for Global Accelerator (%s) deployment: %s", d.Get("accelerator_arn").(string), err)
```

Reply:
This was intended, since the formatted error is now coming from `resourceAwsGlobalAcceleratorAcceleratorWaitForDeployedState`

Reply:
Ah, the waiter function takes the approach of having the error context unlike many others. 👍  

In Go 1.12, this presented the problem of not being able to determine the actual underlying error since the context wrapping would hide it (e.g. you cannot check it with `isResourceTimeoutError()`, since the error type is hidden by the wrapping `fmt.Errorf()`). With Go 1.13 and wrapped errors (`%w` in `fmt.Errorf()`), we likely can start letting functions like these add context messaging themselves, but only assuming they are using `%w`.